### ISSUE TITLE
export.rb:223:in '+' : String can't be coerced into Bignum (TypeError)

### ISSUE THREAD

**Issue by tyrantrex** [zigotica/tumblr-full-backup#2]
I'm not familiar with ruby and needed something to download my photos on tumblr and stumbled across this gem of yours. After fiddling around with the ruby installation and some odd hours of googling, I managed to figure out how to execute scripts. Thinking that everything was going just fine, I ran into a slight issue attached below. Can you please find out what the problem is?

![export rb error](https://cloud.githubusercontent.com/assets/5766919/6022545/25c578f0-ab83-11e4-93f6-173c3ddcd4c6.png)

**Comment by zigotica** [comment]
hi @tyrantrex 

first, thanx for using te script, and even register to open an issue. appreciated. 

5613 posts is definitely too much :D I also have the same problem and still do not know how to handle it (this is my first ruby approach, ever). a good solution might be to also allow for limit/offset to be defined as argument to the script so we can launch it in chunks from the terminal. 

sorry i cannot be much more helpful right now but this is a pet project and i'm really busy these weeks.

**Comment by tyrantrex** [comment]
Thanks for replying so quickly. After playing around with it for a bit, I did manage to get it working with a shoddy workaround. Though it's still running, i'm not sure why it's stuck at 871 files when 1400 have been parsed so far. 

change
```
Line 223: $filename     = $slug"+"_"+i.to_s+"."+$extension
```
to
```
Line 223: $filename     = "#$slug"+"_"+i.to_s+"."+$extension
```

**Comment by zigotica** [comment]
@tyrantrex thanx! seems to work for me, and somewhat makes sense. what i should do is parse the slug to only allow text characters or match a regexp. 

have you been able to download all your 5613 posts?

**Comment by tyrantrex** [comment]
No, but since it is working now, do you mind if I use your script as a base and convert it to another language?

**Comment by zigotica** [comment]
please do, then send me a link so i can check it out, thanx!
also, bear in mind my script is full rubish, my first ruby ever, so expect lots of non-purist stuff, pro ruby coders would probably bleed their eyes watching at this :D

**Comment by zigotica** [comment]
closed by b9fc05a6f5d2830824eccdc404e1446f4c032f5b quick fix
